### PR TITLE
PPS: vector classes used in reco, geometry etc.

### PR DESCRIPTION
**Pull request by jan-kaspar**
#### PR description:

This is a technical PR to reduce the number of different vector classes used in PPS reconstruction, geometry, etc. In particular, the dependence on CLHEP is dropped in favour of ROOT. Several unnecessary vector-format conversion are avoided. In the geometry-consumer code, whenever possible, the auto statement or CTPPSGeometry::Vector typedef is used in order simplify possible future type transitions.

#### PR validation:

Running
```
runTheMatrix.py -l limited --ibeos
```
gave
```
34 33 32 26 16 4 1 1 1 tests passed, 0 0 0 0 0 0 0 0 0 failed
```

This PR does not change the proton reconstruction results - see the comparison below. Blue = before this PR, red dashed = with this PR.
make_cmp

### CODE REVIEW

**Comment by jfernan2** [comment]
please test

**Comment by jfernan2** [comment]
+1
No changes to dqm

**Comment by cvuosalo** [comment]
+1

File: RecoCTPPS/PixelLocal/src/RPixRoadFinder.cc [hunk 1]
```diff
@@ -92,17 +92,17 @@ void RPixRoadFinder::findPattern() {
...
     while (it_gh2 != temp_all_hits.end()) {
       bool same_pot = false;
       CTPPSPixelDetId tmpGh2Id = CTPPSPixelDetId(it_gh2->detId);
       if (currDet.rpId() == tmpGh2Id.rpId())
         same_pot = true;
-      CLHEP::Hep3Vector subtraction = currPoint - it_gh2->globalPoint;
+      const auto subtraction = currPoint - it_gh2->globalPoint;
 
-      if (subtraction.perp() < roadRadius_ && same_pot) {  /// 1mm
+      if (subtraction.Rho() < roadRadius_ && same_pot) {  /// 1mm
```
Comment (perrotta) [inline RecoCTPPS/PixelLocal/src/RPixRoadFinder.cc]: ```perp()``` and ```Rho()``` are not the same: are you fixing a bug here?

**Comment by perrotta** [comment]
@jan-kaspar , although unrelated to this PR, the static analyzer points at what looks like a bug in ```DQM/CTPPS/plugins/CTPPSPixelDQMSource.cc```: at L702 it seems to me that
```C++
        if (clusize > ClusterSizeMax)
          clusize = ClusterSizeMax;
```
should be instead
```C++
        if (clusize > ClusterSizeMax)
          ClusterSizeMax = clusize;
```

Please check, and fix (if you agree with it)

**Comment by jan-kaspar** [comment]
> @jan-kaspar , although unrelated to this PR, the static analyzer points at what looks like a bug in `DQM/CTPPS/plugins/CTPPSPixelDQMSource.cc`: at L702 it seems to me that

@fabferro @AndreaBellora can you please check the suggestion above?

File: RecoCTPPS/PixelLocal/src/RPixRoadFinder.cc (hunk 1 above)
Comment (jan-kaspar) [reply RecoCTPPS/PixelLocal/src/RPixRoadFinder.cc]: Despite the confusing names, perp() in CLHEP and Rho() in ROOT do the same, see e.g. https://root.cern.ch/root/html608/classROOT_1_1Math_1_1DisplacementVector3D.html#a92c6c1ddbb0e06be2aa87e2cc0eb0ddb
I've verified it by explicit print outs.

File: Geometry/VeryForwardRPTopology/interface/RPTopology.h [hunk 2]
```diff
@@ -10,28 +10,21 @@
...
+#include "Math/Vector3D.h"
 
 /**
  *\brief Geometrical and topological information on RP silicon detector.
  * Uses coordinate a frame with origin in the center of the wafer.
  **/
 class RPTopology {
 public:
+  using Vector = ROOT::Math::DisplacementVector3D<ROOT::Math::Cartesian3D<double>>;
+  ;
```
[the remark this answers is not in the document]
Comment (jan-kaspar) [reply Geometry/VeryForwardRPTopology/interface/RPTopology.h]: Done in 8c7259c

**Comment by perrotta** [comment]
please test

**Comment by AndreaBellora** [comment]
> 
> 
> > @jan-kaspar , although unrelated to this PR, the static analyzer points at what looks like a bug in `DQM/CTPPS/plugins/CTPPSPixelDQMSource.cc`: at L702 it seems to me that
> 
> @fabferro @AndreaBellora can you please check the suggestion above?
As far as I can see, this line has not been introduced by the recent PR (#28714), but nevertheless it looks suspicious.
L699 and L701 seem to perform the same check. The fix would although not work, because ClusterSizeMax is a constexpr and thus can't be modified. 
I would assume that the check against ClusterSizeMax is done to set clusize to the maximum allowed if it gets higher than that. The histogram fill should then performed afterwards, otherwise it makes no sense to assign a new value to the variable that goes immediately out of scope.

The check against cluSizeMax is instead done for "monitoring" (reading the comment), but both cluSizeMax and multHitsMax appear to serve no purpose in the code. I would suggest to remove both variables and the if statement at L665. Furthermore move the if statement at L697 below the one at L 701.

**Comment by jan-kaspar** [comment]
> The check against cluSizeMax is instead done for "monitoring" (reading the comment), but both cluSizeMax and multHitsMax appear to serve no purpose in the code. I would suggest to remove both variables and the if statement at L665. Furthermore move the if statement at L697 below the one at L 701.

Thanks @AndreaBellora ! Also to @perrotta , do you prefer to make these modifications in this PR or a dedicated one? Doing it here somewhat kills the "no change" character of this PR.

**Comment by AndreaBellora** [comment]
> 
> 
> > The check against cluSizeMax is instead done for "monitoring" (reading the comment), but both cluSizeMax and multHitsMax appear to serve no purpose in the code. I would suggest to remove both variables and the if statement at L665. Furthermore move the if statement at L697 below the one at L 701.
> 
> Thanks @AndreaBellora ! Also to @perrotta , do you prefer to make these modifications in this PR or a dedicated one? Doing it here somewhat kills the "no change" character of this PR.

In case of a new PR I can follow with that. The change should be pretty quick.

**Comment by perrotta** [comment]
> > > The check against cluSizeMax is instead done for "monitoring" (reading the comment), but both cluSizeMax and multHitsMax appear to serve no purpose in the code. I would suggest to remove both variables and the if statement at L665. Furthermore move the if statement at L697 below the one at L 701.
> > 
> > 
> > Thanks @AndreaBellora ! Also to @perrotta , do you prefer to make these modifications in this PR or a dedicated one? Doing it here somewhat kills the "no change" character of this PR.
> 
> In case of a new PR I can follow with that. The change should be pretty quick.

Yes, please make another "DQM only" PR.
I only wanted to report about my observation here.

**Comment by jan-kaspar** [comment]
> > ...
> > In case of a new PR I can follow with that. The change should be pretty quick.
> 
> Yes, please make another "DQM only" PR.
> I only wanted to report about my observation here.

Thanks to both Andreas!

**Comment by jfernan2** [comment]
+1

**Comment by perrotta** [comment]
+1
- Technical pull request which cleans up and make uniform use of several vector classes in PPS reconstruction
- No changes in reco outputs are expected, and none is observed, also in the private dedicated validation reported in the PR description

**Comment by perrotta** [comment]
@cvuosalo you already signed for geometry, and the last update only removed one useless empty line

**Comment by cvuosalo** [comment]
+1

**Comment by silviodonato** [comment]
+1

**Pull request merged**